superset/datasets/api.py

```diff
@@ -207,7 +207,7 @@ def post(self) -> Response:
         except DatasetInvalidError as ex:
             return self.response_422(message=ex.normalized_messages())
         except DatasetCreateFailedError as ex:
-            logger.error(
+            logger.info(
```

Comment:
If this raises, it's supposed to be a really bad thing, why do you want to lower it, and on what occasions are you getting this?

Reply:
I think we were seeing some instances of dataset creation failure, specifically the unique constraint violation for table_name. Since this error is already bubbled up to the user, and is likely to occur fairly often, I think it makes sense to reduce the noise around this. The fix is likely to add another case for the unique violation, instead of converting all of these to info.